packages/aws-rfdk/lib/deadline/test/worker-fleet.test.ts

```diff
@@ -1884,3 +1886,58 @@ describe('tagging', () => {
     },
   });
 });
+
+test('worker fleet signals when non-zero minCapacity', () => {
+  // WHEN
+  const fleet = new WorkerInstanceFleet(wfstack, 'workerFleet', {
+    vpc,
+    workerMachineImage: new GenericWindowsImage({
+      'us-east-1': 'ami-any',
+    }),
+    renderQueue,
+    minCapacity: 1,
+  });
+
+  // WHEN
+  const userData = fleet.fleet.userData.render();
+
+  // THEN
+  expect(userData).toContain('cfn-signal');
+  expectCDK(wfstack).to(haveResourceLike('AWS::AutoScaling::AutoScalingGroup', {
+    CreationPolicy: {
+      ResourceSignal: {
+        Count: 1,
+      },
+    },
+  }, ResourcePart.CompleteDefinition));
+  // [0] = warning about block devices. [1] = warning about no health monitor
+  expect(fleet.node.metadataEntry).toHaveLength(2);
+});
+
+test('worker fleet does not signal when zero minCapacity', () => {
+  // WHEN
+  const fleet = new WorkerInstanceFleet(wfstack, 'workerFleet', {
+    vpc,
+    workerMachineImage: new GenericWindowsImage({
+      'us-east-1': 'ami-any',
+    }),
+    renderQueue,
+    minCapacity: 0,
+  });
+
+  // WHEN
+  const userData = fleet.fleet.userData.render();
+
+  // THEN
+  // There should be no cfn-signal call in the UserData.
+  expect(userData).not.toContain('cfn-signal');
+  // Make sure we don't have a CreationPolicy
+  expectCDK(wfstack).notTo(haveResourceLike('AWS::AutoScaling::AutoScalingGroup', {
+    CreationPolicy: objectLike({}),
+  }, ResourcePart.CompleteDefinition));
+  // There should be a warning in the construct's metadata about deploying with no capacity.
+  expect(fleet.node.metadataEntry).toHaveLength(3);
+  // [0] = warning about block devices. [2] = warning about no health monitor
+  expect(fleet.node.metadataEntry[1].type).toMatch(ArtifactMetadataEntryType.WARN);
+  expect(fleet.node.metadataEntry[1].data).toMatch(/Deploying with 0 minimum capacity./);
```

Comment:
While the comments help, hard-coding the expected indexes of `metadataEntry` makes the tests fragile. The  assertions depend on side-effects of other behavior in the `WorkerInstanceFleet` construct. I believe we can use the following jest matchers to make this more robust:

*   [`exepect.arrayContaining(...)`](https://jestjs.io/docs/expect#expectarraycontainingarray)
*   [`expect.stringMatching(...)`](https://jestjs.io/docs/expect#expectstringmatchingstring--regexp)

Which would look like:

```ts
expect(fleet.node.metadataEntry).toEqual(expect.arrayContaining([
  {
    type: ArtifactMetaDataEntryType.WARN,
    data: expect.stringMatching(/Deploying with 0 minimum capacity\./),
  },
]));
```

In the case of ensuring the warning is not emitted, we could do:

```ts
expect(fleet.node.metadataEntry).not.toEqual(expect.arrayContaining([
  {
    type: ArtifactMetaDataEntryType.WARN,
    data: expect.stringMatching(/Deploying with 0 minimum capacity\./),
  },
]));
```

This being said, the tests currently pass, so I won't hold up the merge on it. We can change this later if these tests ever break.

Reply:
Oh, I like that!